app/services/xapi/completed_product_service.rb

```diff
@@ -0,0 +1,29 @@
+module Xapi
+  class CompletedProductService < Xapi::BaseService
+    def initialize(agent, course)
+      @agent = agent
+      @object = object(course)
+    end
+
+    def set_statement
```

Comment:
Maybe move this `set_statement` to the `initialize` so we don't need to explicitly call it when using the service.